packages/hub/services/serializers/merchant-info-serializer.ts

```diff
@@ -19,27 +19,56 @@ interface JSONAPIDocument {
 export default class MerchantInfoSerializer {
   databaseManager: DatabaseManager = inject('database-manager', { as: 'databaseManager' });
 
-  async serialize(content: MerchantInfo): Promise<JSONAPIDocument> {
-    const did = encodeDID({ type: 'MerchantInfo', uniqueId: content.id });
-
-    const data = {
-      id: content.id,
-      type: 'merchant-infos',
-      attributes: {
-        did,
-        name: content.name,
-        slug: content.slug,
-        color: content.color,
-        textColor: content.textColor,
-        ownerAddress: content.ownerAddress,
+  async serialize(content: MerchantInfo | string): Promise<JSONAPIDocument> {
+    let data: MerchantInfo;
+    let did: string;
+
+    if (typeof content === 'string') {
+      did = encodeDID({ type: 'MerchantInfo', uniqueId: content });
+      data = await this.loadMerchantInfo(content);
```

Comment:
I would prefer for the serializer to have a narrower responsibility and more constrained constructor parameters by excluding data loading from it's responsibility.

Reply:
Makes sense. I took some inspiration from how the `PrepaidCardCustomizationSerializer` loads the data but I agree it would be good to let the serializer do only the JSON API serialization work. I'm thinking about moving the loading part in the worker task